### DISCUSSION TITLE
Can I use Relm4 in a closed source application?

### DISCUSSION THREAD

**Discussion by Luxvao** [Relm4/Relm4#388] [Q&A] [answered]
I get really confused with licenses. This project is under MIT and Apache 2.0 , which both allow usage in closed source projects. But this project is based on gtk-rs and gtk-rs are the bindings for the GTK library, which is under LGPL. As far as I know LGPL also alows closed source applications, but which license actually applies (LGPL, MIT or Apache 2.0)?
Thanks

**Comment by AaronErhardt** [comment, marked as answer]
Relm4 and gtk-rs choose MIT / Apache to basically get out of your way. The only thing that's relevant for you is the license of GTK which means that you can use it in closed source applications.

Btw. the reason why Relm4 and gtk-rs have different licenses is because statically linking to LGPL isn't allowed for proprietary apps (which is what Rust does) whereas dynamically linking is allowed (which is what is done for GTK).

**Reply by Luxvao** [reply]
So do gtk-rs and relm4 statically link even though GTK is under LGPL? Does the GTK license even matter?

**Reply by AaronErhardt** [reply]
Yes, in fact only the GTK license matters because MIT and Apache are so permissive. Rust will link your application into one binary that includes all crates, but this should only include MIT or Apache licensed crates in your case. GTK is only dynamically linked at runtime, which makes it work with LGPL even for proprietary apps.

**Reply by Luxvao** [reply]
Oh okay
